src/components/welcome/MyAccounts/useAllSafes.ts

```diff
@@ -54,8 +54,10 @@ const useAllSafes = (): SafeItems => {
       const uniqueAddresses = uniq(addedOnChain.concat(ownedOnChain)).filter(Boolean)
 
       return uniqueAddresses.map((address) => {
+        const owners = allAdded?.[chainId]?.[address]?.owners
+        const isOwner = owners?.map((address) => address.value).includes(walletAddress)
```

Comment:
```suggestion
        const isOwner = owners?.some(({ value }) => sameAddress(walletAddress))
```

There can be a mismatch in lowercase vs checksummed address format otherwise.